sw/host/opentitanlib/src/spiflash/flash.rs

```diff
@@ -151,12 +151,19 @@ impl SpiFlash {
         loop {
             // READ_SFDP always takes a 3-byte address followed by a dummy byte regardless of
             // address mode.
-            spi.run_eeprom_transactions(&mut [Transaction::Read(
-                MODE_111
-                    .dummy_cycles(8)
-                    .cmd_addr(SpiFlash::READ_SFDP, 0, AddressMode::Mode3b),
-                &mut buf,
-            )])?;
+            let mut eeprom_transactions: Vec<Transaction> = Vec::new();
+            let read_size = spi.get_eeprom_max_transfer_sizes()?.read;
```

Comment:
What is the difference between max_transfer_size and eeprom_max_transfer_size?

I'm thinking about your forthcoming dediprog implementation, where the max_transfer_size is something like the capability of the transceive function, whereas the eeprom_max_transfer_size is the capability of the specialized bulk read/write functions.  If I'm modelling it correctly in my mind, then the read size should be max_transfer_size, correct?

Reply:
Your understanding is generally correct.  `get_max_transfer_size()` is meant to advertise the maximum valid inputs to `run_spi_transaction()` , and `get_eeprom_max_transfer_size()` is meant to advertise the maximum valid inputs to `run_eeprom_transactions()`.

In this piece of code, the latter `run_eeprom_transactions()` is being used, but the opcode is not that of a standard read, but instead the "Read SFDP" opcode.  So on the face of it, it is correct to use `get_eeprom_max_transfer_sizes()`.

Actual implementations are complicated, and the limitations cannot be summarized into four numbers. Dediprog as you mention have such limitations.   I have not been able to make it send arbitrary read opcodes even when setting byte 4 while using 4B_ADDR_FAST_0x0C, so it will not be able to support the same length of reading SFDP as reading actual data.

To compensate for that, and maintain the contract that `run_eeprom_transactions()` can take any reads up to the advertized size, I have some workaround logic inside the Dediprog driver, which recognizes the "Read SFDP" opcode, and breaks it down into several 16-byte operations behind the scenes.

Reply:
I don't expect reading the SFDP to ever work via the bulk pipes on dediprog; the correct chunk size for this command on that device is 16 bytes.  It sounds like you're treating it as a special case within the dediprog driver.